### PR TITLE
Fast optimization mode

### PR DESCRIPTION
**Pull request by igordertigor**
This pull request contains four things:

1. I'm including code that allows for a quick and dirty analysis of what had been there before. This is mostly covered in the first commit. There is a lot of code that I'm removing, mostly because it didn't work with the new structure anymore.
2. I'm adding a bit more documentation about fitting psychometric functions in the readme
3. I increase unit test coverage to above 80% because that's policy on this repository.
4. I make sure that the doctests run with the unittests

Apologies for mixing all this up, I just wanted to get it done as soon as possible.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,3 +22,51 @@ out their documentation for how to build and install it.
...
+...                  [0.6, 1.0, 5]])
+>>> params, ll = psychometric.fit(data)
+
+```
+
+Note that the data format has stimulus intensity in the first column, then
+the fraction of correct responses and then the number of responses
+collected.
+
+While we connect data, it is often useful to have a quick and dirty way to
```
Comment (JaykishanPatel) [inline README.md]: Did you mean collect?

File: src/main/python/esdt/pmf.py [hunk 2]
```diff
@@ -1,69 +1,175 @@
...
+        else:
+            raise RuntimeError(
+                'Trying to use pylab function by pylab does not exist')
+    col = kwargs.setdefault('color', 'black')
+
+    dataset = pmf.extract_data()
+    xmin = dataset.x.min()
+    xmax = dataset.x.max()
+    x = kwargs.pop('x', np.mgrid[xmin:xmax:100j])
+    i0 = x <= xmin
```
Comment (JaykishanPatel) [inline src/main/python/esdt/pmf.py]: I seem to have changed the color of this line to yellow, i'm not sure what that means but there is nothing wrong with this line.

File: src/main/python/esdt/pmf.py [hunk 3]
```diff
@@ -1,69 +1,175 @@
...
+    i1 = x >= xmax
+    i2 = np.logical_and(x >= xmin, x <= xmax)
```
Comment (JaykishanPatel) [inline src/main/python/esdt/pmf.py]: This just returns an array the size of ```x``` with all ```true``` values, right?

File: src/unittest/python/pmf_tests.py [hunk 4]
```diff
@@ -9,68 +9,68 @@
...
 
     def setUp(self):
-        self.data = np.array([[0.4, 1, 5],
-                              [0.8, 3, 5],
-                              [1.2, 5, 5],
-                              [1.3, 4, 5]])
+        self.data = np.array([[0.4, 0.2, 5],
+                              [0.8, 0.6, 5],
+                              [1.2, 1.0, 5],
+                              [1.3, 0.8, 5]])
```
Comment (JaykishanPatel) [inline src/unittest/python/pmf_tests.py:15]: So from what i can gather, these changes affect the ```proportion correct``` parameter. I would like to know why there is a 1.0 before the 0.8 in the matrix though.  

```python
self.data = np.array([[0.4, 0.2, 5],
                                 [0.8, 0.6, 5],
                                 [1.2, 1.0, 5],
                                 [1.3, 0.8, 5]])
```

**Review by JaykishanPatel — APPROVED** [review]
The unit test are still somewhat above my grasp, but other than that, I think, i've mentioned everything else that could be fixed. I also wanted to mention that we should fix the missing last row problem soon, and maybe add that to the unit tests?

File: README.md (hunk 1 above)
Comment (igordertigor) [reply README.md]: Yes. fixed by commit message `connect -> collect`

File: src/main/python/esdt/pmf.py (hunk 3 above)
Comment (igordertigor) [reply src/main/python/esdt/pmf.py]: No. There are three ranges here: `i0` is values below the range of recorded data, `i1` is above recorded data, `i2` is the range in which the psychometric function is actually supported by data. I changed the names in commit `more meaningful names in plotting`.

File: src/unittest/python/pmf_tests.py (hunk 4 above)
Comment (igordertigor) [reply src/unittest/python/pmf_tests.py:15]: Not sure what you're asking here. Which one? In the third row? What's wrong with that?

Comment (JaykishanPatel) [reply src/unittest/python/pmf_tests.py:15]: In the matrix array attached to the first comment the middle column has numbers 0.2, 0.4, 1.0, 0.8 in that order. I was wondering why 0.8 comes after 1.0. Hopefully that’s more clear

Comment (igordertigor) [reply src/unittest/python/pmf_tests.py:15]: I think I understand what you mean. You would have expected that the 0.8 should come before the 1.0, right? Let's assume our psychometric function at intensity 0.4 is 0.5, at 0.8 it rises to, say 0.75 and after 1.1 it is more or less constant at 0.95, ok? Now the last column says that all these intensities, we collected 5 trials each. Then the 0.2 (first row) corresponds to 1 out of 5 correct (with a probability for correct of 0.5 the likelihood for that is about 15.6%), the 0.6 (second row) corresponds to 3 out of 5 correct (with probability for correct being 0.75, that is 26.4% likely), the third row says that 5 out of 5 were correct (with probability for correct of 0.95 that is 77.4% likely), and the last row says that 4 out of 5 were correct at a stimulus intensity of 1.3 (with probability for correct of 0.95 that is still 20.3% likely). So none of these is a particularly surprising result. Does that answer your question?

Comment (JaykishanPatel) [reply src/unittest/python/pmf_tests.py:15]: Yes! My understanding was that the numbers represented probability and not trial results. Thank you.

**Pull request merged**